quail/processes/wps_climdex_temp_pctl.py

```diff
@@ -0,0 +1,143 @@
+    - climdex.tx90p: computes the monthly or annual percent of values above the 90th percentile of baseline daily maximum temperature.
+    """
+
+    def __init__(self):
+        self.status_percentage_steps = dict(
+            common_status_percentages,
+            **{
+                "load_rdata": 10,
+                "save_rdata": 90,
+            },
+        )
+        inputs = [
+            climdex_input,
+            ci_name,
+            output_file,
+            LiteralInput(
+                "func",
+                "Function to compute",
+                abstract="Percentile function to compute",
+                allowed_values=["tn10p"],
+                min_occurs=1,
+                max_occurs=1,
+                data_type="string",
+            ),
+            LiteralInput(
+                "freq",
+                "Time frequency",
+                abstract="Time frequency to aggregate to",
+                allowed_values=["monthly", "annual"],
+                data_type="string",
+            ),
+            vector_name,
+            log_level,
+        ]
+
+        outputs = [rda_output]
+
+        super(ClimdexTempPctl, self).__init__(
+            self._handler,
+            identifier="climdex_temp_pctl",
+            title="Climdex Temperature Percentiles",
+            abstract="Percent of Values Above/Below 10th/90th Percentile Daily Maximum/Minimum Temperature",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def _handler(self, request, response):
+        climdex_input, ci_name, output_file, func, freq, vector_name, loglevel = [
+            arg[0] for arg in collect_args(request, self.workdir).values()
+        ]
+
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+        robjects.r("library(climdex.pcic)")
+
+        log_handler(
+            self,
+            response,
+            "Loading climdexInput from R data file",
+            logger,
+            log_level=loglevel,
+            process_step="load_rdata",
+        )
+        ci = load_rdata_to_python(climdex_input, ci_name)
+
+        log_handler(
+            self,
+            response,
+            f"Processing climdex.{func}",
+            logger,
+            log_level=loglevel,
+            process_step="process",
+        )
+
+        robjects.r.assign("freq", freq)
+        mothly_pct = robjects.r(f"climdex.{func}(ci, freq)")
```

Comment:
You don't have to use the assign here for freq you can just use an f string

`mothly_pct = robjects.r(f"climdex.{func}(ci, {freq})")`

The only reason I did this in my other process is because Python boolean `True/ False` does not translate directly to R logical `TRUE/FALSE` or `T/F`

Reply:
@cairosanders Actually, a single quote should wrap `{freq}` to make the function work. However, I guess `robjects.r.assign("freq", freq)` converts python string to R string so that the single quote is not necessary. It seems like using the `robjects.r.assign` for the string inputs is more of a standardized way. What do you think?

Reply:
Oh ya sorry, you do need the the quotes around it as well sorry about that. I  think the assign is an extra unnecessary step, I don't think it's a big deal to just use the quotes.